File: benches/cpp/fibonacci.cpp
// Fibonacci benchmark for C++

#include <iostream>
#include <cstdlib>
#include <cstdint>

int64_t fib(int64_t n) {
    if (n <= 1) {
        return n;
    }
    return fib(n - 1) + fib(n - 2);
}

int main(int argc, char* argv[]) {
    int64_t n = argc > 1 ? std::atoll(argv[1]) : 20;
    int64_t result = fib(n);
    std::cout << result << std::endl;
    return 0;
}

File: benches/python/fibonacci.py
#!/usr/bin/env python3
"""Fibonacci benchmark for Python"""

def fib(n: int) -> int:
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

def main():
    import sys
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 20
    result = fib(n)
    print(result)

if __name__ == "__main__":
    main()

